st2common/st2common/transport/consumers.py

```diff
@@ -86,3 +107,33 @@ def shutdown(self):
     @abc.abstractmethod
     def process(self, message):
         pass
+
+    def _get_queue_consumer(self, connection, queues):
+        return QueueConsumer(connection, queues, self)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class StagedMessageHandler(MessageHandler):
+    """
+    MessageHandler to deal with messages in 2 steps.
+        1. pre_ack_process : This is called on the handler before ack-ing the message.
+        2. process: Called after ack-in the messages
```

Comment:
Honestly, for consistency and to make it more obvious, I would prefer `process` to be called `post_ack_process` (or similar), but I imagine you just kept it as `process`, because it was called that before.